sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -379,6 +381,33 @@ class Analyzer(catalog: Catalog,
     }
   }
 
+  object ResolveSortAlias extends Rule[LogicalPlan] {
```

Comment:
I think that it is better to logically group all of the special handling for ORDER BY clauses into a single rule.